### PR TITLE
New Website: Add Aria Labels and Better Alt Text

### PR DESCRIPTION
**Pull request by andrew032011**
### Summary 
- Add Descriptive Aria Labels where appropriate
- Add and revise alt text for images to be more descriptive

### Notion/Figma Link 
Addresses problem 3 and problem 6 from https://github.com/cornell-dti/idol/issues/734.

https://www.notion.so/Address-Code-Audit-Comments-14b0ad723ce1802fa2f0c633511d8de3?pvs=4

### Test Plan 
No UI Changes expected. Grep through the code base for all instances of aria-label, alt, Image, img, button, and Link to ensure that the proper aria labels and alt texts are provided.

### CODE REVIEW

File: new-dti-website/components/apply/ApplyFAQ.tsx [hunk 1]
```diff
@@ -26,6 +26,7 @@ const ApplyFAQ = () => {
           section === question ? 'text-[#FEFEFE] bg-[#0C0404]' : ''
         }`}
       onClick={() => setQuestion(section)}
+      aria-label={`select ${section} section`}
```
Comment (Bookie0) [inline new-dti-website/components/apply/ApplyFAQ.tsx:29]: yay for aria labels!

Just a small nit i found when testing – do you think we could make the `<button>` wrap the label as well ("Design", "Business", etc.)? That way you can click on the entire container (label + icon).

It's just a smol polish thing that could increase the tap target and make it easier to select the tabs


Screenshot 2024-11-30 at 12 41 14 AM

File: new-dti-website/components/bottom.tsx [hunk 2]
```diff
@@ -85,22 +85,22 @@ const Bottom: React.FC = () => (
...
-            alt="outreach 3"
+            alt="kids being mentored by DTI member"
           />
           <Image
             className="w-1/3 self-start"
             src="/images/outreach_2.svg"
             width={115}
             height={103}
-            alt="outreach 2"
+            alt="students learning Figma from DTI members"
```
Comment (Bookie0) [inline new-dti-website/components/bottom.tsx:103]: yess ty  for this

File: new-dti-website/components/bottom.tsx [hunk 3]
```diff
@@ -128,8 +128,8 @@ const Bottom: React.FC = () => (
         src="/images/team.svg"
         width={377}
         height={286}
-        alt="team"
-      ></Image>
+        alt="team photo"
```
Comment (Bookie0) [inline new-dti-website/components/bottom.tsx:131]:
```suggestion
        alt="team photo of DTI members"
```

File: new-dti-website/components/course/DDProjects.tsx [hunk 4]
```diff
@@ -35,6 +35,7 @@ export default function DDProjects({ title, description, imageSrc }: DDProjectsP
         isOpen ? 'bg-red-500' : 'bg-white'
       } w-full max-w-8xl rounded-xl drop-shadow-sm px-10 py-8 border-1 border-[#E4E4E4]`}
       onClick={toggleCard}
+      aria-label="toggle card"
```
Comment (Bookie0) [inline new-dti-website/components/course/DDProjects.tsx:38]: could we possibly make the alt text dynamic based on what the project is actually called?

ie: "Expand todo list app student project" or "Minimize todo list app student project"

File: new-dti-website/components/team/MemberGroup.tsx [hunk 5]
```diff
@@ -343,7 +350,11 @@ const MemberGroup: React.FC<MemberGroupProps> = ({
         >
           {members.map((member, index) => (
             <>
-              <button onClick={() => onMemberCardClick(member)} className="memberCard">
+              <button
+                onClick={() => onMemberCardClick(member)}
+                className="memberCard"
+                aria-label={`open ${member.firstName} ${member.lastName}'s details`}
+              >
```
Comment (Bookie0) [inline new-dti-website/components/team/MemberGroup.tsx:357]: yes yes yes ty for this!

File: new-dti-website/components/bottom.tsx [hunk 6]
```diff
@@ -85,22 +85,22 @@ const Bottom: React.FC = () => (
           src="/images/outreach_1.svg"
           width={275}
           height={143}
-          alt="outreach 1"
+          alt="kids stacking boxes at make-a-thon"
```
Comment (Bookie0) [inline new-dti-website/components/bottom.tsx:88]: while we're at it adding alt text, do u think u could add alt text for the images on the Initiatives page? i think the images a missing a few or aren't too precise:


Screenshot 2024-11-30 at 12 46 53 AM
Screenshot 2024-11-30 at 12 47 06 AM

Screenshot 2024-11-30 at 12 47 23 AM

Comment (Bookie0) [reply new-dti-website/components/bottom.tsx:88]: Some of my suggestions (feel free to edit as u see fit):

|Image|Alt text|
|-|-|
|image|"a DTI member teaching members of the Ithaca community about technology"|
|image|"a DTI member presenting a series of graphs at the Half-Baked event"|
|image|"a DTI member presenting code at the BigRed//Hacks hackathon event"|
|image|"DTI members posing in front of the Millennium logo"|

**Review by Bookie0 — APPROVED** [review]
Awesome! thank you andrew

a few comments, but feel free to take or leave them :)

File: new-dti-website/components/apply/RoleDescription.tsx [hunk 7]
```diff
@@ -29,7 +29,12 @@ const RoleDescriptions = () => {
           {Object.keys(applications).map((application) => {
             const { icon } = applications[application];
             return (
-              <div className="flex flex-col items-center gap-7" key={application}>
+              <button
+                onClick={() => setRole(application)}
+                aria-label={`Show ${application} questions`}
+                className="flex flex-col items-center gap-7"
+                key={application}
+              >
```
Comment (Bookie0) [inline new-dti-website/components/apply/RoleDescription.tsx:37]: great, ty for this!

File: new-dti-website/components/course/DDProjects.tsx [hunk 8]
```diff
@@ -35,7 +35,7 @@ export default function DDProjects({ title, description, imageSrc }: DDProjectsP
         isOpen ? 'bg-red-500' : 'bg-white'
       } w-full max-w-8xl rounded-xl drop-shadow-sm px-10 py-8 border-1 border-[#E4E4E4]`}
       onClick={toggleCard}
-      aria-label="toggle card"
+      aria-label={`${isOpen ? 'Minimize' : 'Expand'} ${title} student project`}
```
Comment (Bookie0) [inline new-dti-website/components/course/DDProjects.tsx:38]: yessss awesome

File: new-dti-website/components/initiatives/data/initiatives.json [hunk 9]
```diff
@@ -10,7 +10,7 @@
     },
     "image": {
       "src": "/images/tcpl.jpg",
-      "alt": "tcpl",
+      "alt": "a DTI member teaching members of the Ithaca community about technology at the Tompkins County Public Library",
       "width": 563,
```
Comment (Bookie0) [inline new-dti-website/components/initiatives/data/initiatives.json:14]: 🤩 ty!

**Review by Bookie0 — APPROVED** [review]
wooh great changes thanks andrew!

**Review by JasonMun7 — APPROVED** [review]
This looks great! Will help a lot in the future for whenever we inspect the element for any bugs! I think in the future I would like to reuse the Button component in components/ui/button.tsx for all these `<button>` tags but definitely can be addressed in a future PR!

**Comment by andrew032011** [comment]
> This looks great! Will help a lot in the future for whenever we inspect the element for any bugs! I think in the future I would like to reuse the Button component in components/ui/button.tsx for all these `<button>` tags but definitely can be addressed in a future PR!

Filed a ticket for this: https://www.notion.so/Replace-button-components-with-Button-1520ad723ce1808e991bdc1731daa82e?pvs=4

**Pull request merged**